### PR TITLE
📖  updates to release task docs and notes tool

### PR DESCRIPTION
**Pull request by joekr**
**What this PR does / why we need it**:
While going through a release some context was missing in docs.

The comms team updates the `Fun stats` to `Overview` manually,  this will make the job a bit more automated.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
No open issue

### CODE REVIEW

File: docs/release/release-tasks.md [hunk 1]
```diff
@@ -188,6 +188,9 @@ to a newer Go minor version according to our [backport policy](./../../CONTRIBUT
 1. Ensure CI is stable before cutting the release (e.g. by checking with the CI manager)
    Note: special attention should be given to image scan results, so we can avoid cutting a release with CVE or document known CVEs in release notes.
 2. Create and push the release tags to the GitHub repository:
+
+   **NOTE:** when cutting a release tag it will cause issues for users trying to download the latest release tag. See [issue 7889](https://github.com/kubernetes-sigs/cluster-api/issues/7889) for more details
```
Comment (killianmuldoon) [inline docs/release/release-tasks.md:192]: Can we add that this is resolved when the release is published? e.g. 

```suggestion
   **NOTE:** clusterctl will have issues installing providers between the time the release tag is cut and the Github release is published. See [issue 7889](https://github.com/kubernetes-sigs/cluster-api/issues/7889) for more details.
```

**Review by killianmuldoon — COMMENTED** [review]
Looks good in general - has someone got a todo to do a follow up for the release notes generation tool? i.e. adding a make target and ensuring the same binary is run on each release branch?

**Comment by joekr** [comment]
Yes, I've noted the "release notes tool" update in the comms list here https://github.com/kubernetes-sigs/cluster-api/issues/8418

File: docs/release/release-tasks.md (hunk 1 above)
Comment (joekr) [reply docs/release/release-tasks.md]: updated.

File: docs/release/release-tasks.md [hunk 2]
```diff
@@ -215,6 +218,7 @@ to a newer Go minor version according to our [backport policy](./../../CONTRIBUT
         * `kpromo` uses `git@github.com:...` as remote to push the branch for the PR. If you don't have `ssh` set up you can configure
           git to use `https` instead via `git config --global url."https://github.com/".insteadOf git@github.com:`.
         * This will automatically create a PR in [k8s.io](https://github.com/kubernetes/k8s.io) and assign the CAPI maintainers.
+        * Wait for the [promotion prow job](https://prow.k8s.io/?repo=kubernetes%2Fk8s.io&job=post-k8sio-image-promo) to complete successfully before going to the next step.
```
Comment (sbueringer) [inline docs/release/release-tasks.md:221]: I think this should happen after the merge. Merge is step 4

Comment (joekr) [reply docs/release/release-tasks.md:221]: great point. I will do that.

**Review by fabriziopandini — COMMENTED** [review]
/lgtm

**Review by killianmuldoon — COMMENTED** [review]
/lgtm
/approve

**Comment by killianmuldoon** [comment]
No need to backport this IMO given it's not in the book. Release process should always be followed on main - otherwise the end result is following 2-3 release processes each time with slight variation.

**Pull request merged**